src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1383,11 +1503,34 @@ private void WriteRedListHelperMethods(Node node)
             }
         }
 
+        private Node TryGetNodeForNestedList(Field field)
+        {
+            Node referencedNode = GetNode(field.Type);
+            if (referencedNode != null && (!IsOptional(field) || RequiredFactoryArgumentCount(referencedNode) == 0))
+            {
+                return referencedNode;
+            }
+
+            return null;
+        }
+
         private void WriteRedListHelperMethods(Node node, Field field)
         {
             var argType = GetElementType(field.Type);
+
+            var isNew = false;
+            if (IsOverride(field))
+            {
+                var baseType = GetHighestBaseTypeWithField(node, field.Name);
+                if (baseType != null)
+                {
+                    WriteLine("    internal override {0} Add{1}Core(params {2}[] items) => Add{1}(items);", baseType.Name, field.Name, argType);
+                    isNew = true;
+                }
+            }
+
             WriteLine();
-            WriteLine("    public {0} Add{1}(params {2}[] items)", node.Name, field.Name, argType);
+            WriteLine($"    public {(isNew ? "new " : " ")}{node.Name} Add{field.Name}(params {argType}[] items)");
```

Comment:
>  [](start = 55, length = 1)

I think this space (" ") is the culprit to noisy diff.

Reply:
Yup.  Will fix.

Reply:
Done.